builder/builder-next/controller.go

```diff
@@ -537,13 +537,17 @@ func parseGCPolicy(p config.BuilderGCRule, prefix string) (reservedSpace, maxUse
 
 func getEntitlements(conf config.BuilderConfig) []string {
 	var ents []string
-	// Incase of no config settings, NetworkHost should be enabled & SecurityInsecure must be disabled.
+	// In case of no config settings, NetworkHost and Device should be enabled
+	// but SecurityInsecure must be disabled.
 	if conf.Entitlements.NetworkHost == nil || *conf.Entitlements.NetworkHost {
 		ents = append(ents, string(entitlements.EntitlementNetworkHost))
 	}
 	if conf.Entitlements.SecurityInsecure != nil && *conf.Entitlements.SecurityInsecure {
 		ents = append(ents, string(entitlements.EntitlementSecurityInsecure))
 	}
+	if conf.Entitlements.Device == nil || *conf.Entitlements.Device {
+		ents = append(ents, string(entitlements.EntitlementDevice))
+	}
```

Comment:
Does BuildKit have some list of its defaults? Slightly starting to wonder if we need to change the `BuilderEntitlements` to be a `map[string]bool` (or something) that allows it to stay in sync with whatever new options are added in BuildKit

That may also allow us to validate options as part of `dockerd --validate` (i.e., check the list of keys against known entitlements in BuildKit)

Reply:
> Does BuildKit have some list of its defaults?

They are all disabled by default in buildkit

> Slightly starting to wonder if we need to change the `BuilderEntitlements` to be a `map[string]bool` (or something) that allows it to stay in sync with whatever new options are added in BuildKit

Yes I think we could but needs to keep current defaults for moby.

Reply:
Yeah, considering we could set some defaults here (which is where we set most defaults); https://github.com/moby/moby/blob/d8089c82edda48a773ca202f54d29246f6cf4dd0/daemon/config/config.go#L311-L345


We may need to look closely if we want to import BuildKit packages there, or define local consts for the defaults (but possibly it's fine to import buildkit code).